criu/files.c

```diff
@@ -1346,6 +1346,24 @@ static int fchroot(int fd)
 	return chroot(".");
 }
 
+static int need_chroot(int saved_root)
+{
+	struct stat saved_root_stat, cur_root_stat;
+
+	if (fstat(saved_root, &saved_root_stat) == -1) {
+		pr_perror("Failed to stat saved root dir");
+		return -1;
+	}
+
+	if (stat("/proc/self/root", &cur_root_stat) == -1) {
```

Comment:
pls use open_pid_proc, open_proc, open_proc_rw to access proc files.
```
psd = open_pid_proc(PROC_SELF);
if (fstatat(psd, "root", &cur_root_stat) == -1)  {
...
```

Reply:
Fixed, thanks.